internal/bin/sql/sqlgen.go

```diff
@@ -0,0 +1,371 @@
+// (c) Copyright IBM Corp. 2022
+//go:build go1.18
+
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	combinations "github.com/mxschmitt/golang-combinations"
+)
+
+const driverStmt = "driver.Stmt"
+
+var _conn_m = map[string]string{}
+var _stmt_m = map[string]string{}
+
+var arrayConn = []string{
+	"driver.Conn",
+	"driver.Execer",
+	"driver.ExecerContext",
+	"driver.Queryer",
+	"driver.QueryerContext",
+	"driver.ConnPrepareContext",
+	"driver.NamedValueChecker",
+}
+
+var arrayStmt = []string{
+	"driver.Stmt",
+	"driver.StmtExecContext",
+	"driver.StmtQueryContext",
+	"driver.NamedValueChecker",
+	"driver.ColumnConverter",
+}
+
+func inArray(el string, arr []string) bool {
+	for _, v := range arr {
+		if v == el {
+			return true
+		}
+	}
+
+	return false
+}
+
+func removeOnceFromArr(el string, arr []string) []string {
+	for k, v := range arr {
+		if v == el {
+			return append(arr[:k], arr[k+1:]...)
+		}
+	}
+	return arr
+}
+
+func main() {
+
+	fmt.Println("// (c) Copyright IBM Corp. 2022")
+	fmt.Println("// Code generated by sqlgen. DO NOT EDIT.")
+	fmt.Println("")
+	fmt.Println("package instana")
+	fmt.Println("import \"database/sql/driver\"")
+
+	arrayConnCopy := make([]string, len(arrayConn))
+	arrayStmtCopy := make([]string, len(arrayStmt))
+
+	copy(arrayStmtCopy, arrayStmt)
+	copy(arrayConnCopy, arrayConn)
+
+	gen("driver.Conn", "w_conn_", arrayConn)
+	gen(driverStmt, "w_stmt_", arrayStmt)
+
+	printFunctionMaps(arrayConnCopy, arrayStmtCopy)
+	printUtil()
+}
+
+// todo: use copy instead of modify input array
+func gen(basicTypeForWrapper, prefix string, listOfTheOriginalTypes []string) {
+	// generate all subsets of the interfaces
+	conn := combinations.All(listOfTheOriginalTypes)
+	var filteredSubsets [][]string
+
+	for _, v := range conn {
+		if inArray(basicTypeForWrapper, v) {
+			//remove basic type
+			v = v[1:]
+			if len(v) == 0 {
+				continue
+			}
+			filteredSubsets = append(filteredSubsets, v)
+		}
+
+	}
+
+	// not really important
```

Comment:
I think we can remove this comment and keep the sorting, no?
This means that the generated code will look more organized and easier to read, as far as I understand.

Reply:
Let's remove the comment